### DISCUSSION TITLE
Is there any way to access effects['url'] in component's PHP?

### DISCUSSION THREAD

**Discussion by jeffchown** [livewire/livewire#6229] [1. Help]
Hi, all.

Caleb & team, thanks for all your hard work!  My early dive into v3 has been great!

I have one pattern I've been using in a v2 app that I could use help with in v3:

I have multiple $queryString properties that I serialize to a var and store to db for persistence across a user's sessions.

In v2, I simply serialize the $queryString array and store it in my db.

I'd like to convert that logic to the utilization of v3's #[Url] attribute.

As far as I can tell (from Livewire 3's js and test code), the #[Url] properties are stored in a ComponentContext instance and accessed via $this->effects['url'] but I can't find a way to access this is my component's PHP code.

Is there any way I can accomplish the above?

**Comment by joshhanley** [comment]
@jeffchown I'm not recommending this (😉) - but you could look at how the pagination feature talks to the query string feature to get all query string properties https://github.com/livewire/livewire/blob/b103508c0aeacd87efeb57ec6d0c7956fa5b9522/src/Features/SupportPagination/SupportPagination.php#L119-L124

**Comment by jeffchown** [comment]
Recommending what, @joshhanley? ;)

Thanks!  I'll give it a try.

**Comment by jeffchown** [comment]
Morning, @joshhanley 

Thanks, I tried your suggestion but don't think it'll cover all the bases on my end - but, I think I may be overcomplicating things re: my needs...

**Scenario:**

- I have created a Livewire 3 component called DataList
- It renders a collection of data records
- There are a few buttons that (via e.g. `wire:click="setStatus('approved')"`) activate specific filter conditions by updating a `$status` property which of course re-renders the list after performing a new db query that applies the filter condition
- the `$status` property is included in my `$queryString` array (or, hopefully soon, using the #[Url] attribute in v3)
- the browser's URL is updated properly (e.g. to `https://myapp.test/documents?status=approved`

**Goal:**

To save the updated URL to a db record to persist the user's selections between sessions and also facilitate a 'Recents' dropdown in my app.

**Issue/Question:**

I can't find an easy way to get the updated URL after a filter button is pressed as any use of `url()`, `URL::full()`, `request()->url()` (of course) returns `https://myapp.test/livewire/update` when a button is clicked.

What is the best way to get the browser's current & full URL each time Livewire updates the browser's query string? 
(PS I can't wait until the user navigates to another page and then capture the referer url)

Thanks in advance for any help/ideas!

**Reply by jeffchown** [reply]
@joshhanley Or, is there a Livewire internal method or variable that holds the current url (w/ query string) that I can call in an `updated()` method?  Or a way to hook into the query string `updating`/`updated`?

**Reply by jeffchown** [reply]
Hi @joshhanley - do you have any ideas/thoughts re: ☝️?

PS Thanks to you, @calebporzio and the rest of the team for the great work on Livewire 3!
